syntax = "proto3";

message ConjugationInput {
    string root = 1;
    string option = 2;
    string agent = 3;
    string patient = 4;
}

message ResponseMorpheme {
    int32 position = 1;
    string value = 2;
    string gloss = 3;
    string english = 4;
    repeated string type = 5;
}

message ConjugationOutput {
    repeated ResponseMorpheme morphemes = 1;
}

message ResponseObject {
    ConjugationInput input = 1;
    ConjugationOutput output = 2;
}

message Response {
    repeated ResponseObject response = 1;
}